rest_framework/throttling.py

```diff
@@ -120,7 +121,7 @@ def allow_request(self, request, view):
         if self.key is None:
             return True
 
-        self.history = self.cache.get(self.key, [])
+        self.history = self.cache.get(self.key, deque())
```

Comment:
Will constructing a `deque()` only to be thrown away when hitting the cache be a problem?

Reply:
Hi @akx,  

> Will constructing a deque() only to be thrown away when hitting the cache be a problem?

No, the initialization way is the same as the empty list version. As you can see, there is no other way to initialize `self.history` but Line 124. What we only modify is the way to maintain a FILO array (`self.history`) by using `collections.deque` which is most suitable for the scenario.

Since we replace the data structure of `SimpleThrottleRate` that might be the most common throttle class, we must clarify it in the changelog to avoid the upgrade defect.

Reply:
My point was mostly that is it e.g. slower to initialize a `deque` than an empty list? Should this explicitly be

```suggestion
        self.history = self.cache.get(self.key)
        if self.history is None:
            self.history = deque()
```
?

Reply:
I just check the [cpython/listobject.c](https://github.com/python/cpython/blob/main/Objects/listobject.c) and [cpython/Modules/_collectionsmodule.c](https://github.com/python/cpython/blob/main/Modules/_collectionsmodule.c), and  [django-redis/cache.py](https://github.com/jazzband/django-redis/blob/master/django_redis/cache.py) since `django/core/cache/backend/base.py` wants us to implement `.get()` ourselves.

The initialization performance difference between `list` and `deque` is close I think, but seems like we will declare a `deque()` in every `allow_request()` no matter the key exists in cache or not, so I think the explicit way like yours will be better than the original version.

Reply:
A simple benchmark:
```python
In [1]: timeit.timeit("[]", number=10000)
Out[1]: 0.00045701999999891996

In [2]: timeit.timeit("deque()", setup="from collections import deque", number=10000)
Out[2]: 0.000674222999975882
```